### PR TITLE
[BUGFIX] incorrect cache of pull plan with mixed builtin + user functions

### PR DESCRIPTION
**Pull request by Ignition**
Main fix is for incorrect pull plan caching when query contained both builtin and user functions.

Also minor changes:
- Fix for operator== of WalDeltaData, only used in tests, but was wrong
- Moved memory ownership out of operator and into cursor (TODO: in another PR remove this dedicated memory resource)
- Deleted dead code bond.hpp

### CODE REVIEW

**Comment by Ignition** [comment]
### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

File: src/query/frontend/ast/cypher_main_visitor.cpp [hunk 1]
```diff
@@ -2893,7 +2893,9 @@ antlrcpp::Any CypherMainVisitor::visitFunctionInvocation(MemgraphCypher::Functio
   // repeativly finding the function at call time, this means user-defined functions have there lifetime
   // tied to the ast Function operation. We do not want that cached, because we want to be able to reload
   // query module that is not currently being used.
-  query_info_.is_cacheable = !function_expr->IsUserDefined();
+  if (function_expr->IsUserDefined()) {
+    query_info_.is_cacheable = false;
```
Comment (andrejtonev) [inline src/query/frontend/ast/cypher_main_visitor.cpp:2897]: We can use &= instead of the if

File: src/storage/v2/durability/wal.cpp [hunk 2]
```diff
@@ -610,14 +610,13 @@ bool operator==(const WalDeltaData &a, const WalDeltaData &b) {
...
-    case WalDeltaData::Type::LABEL_PROPERTY_INDEX_DROP:
     case WalDeltaData::Type::TEXT_INDEX_CREATE:
-      return a.operation_text.index_name == b.operation_text.index_name &&
-             a.operation_text.label == b.operation_text.label;
     case WalDeltaData::Type::TEXT_INDEX_DROP:
       return a.operation_text.index_name == b.operation_text.index_name &&
              a.operation_text.label == b.operation_text.label;
+
+    case WalDeltaData::Type::LABEL_PROPERTY_INDEX_CREATE:
+    case WalDeltaData::Type::LABEL_PROPERTY_INDEX_DROP:
```
Comment (andrejtonev) [inline src/storage/v2/durability/wal.cpp:619]: So now the index name is irrelevant?

**Review by andrejtonev — APPROVED** [review]

File: src/storage/v2/durability/wal.cpp (hunk 2 above)
Comment (Ignition) [reply src/storage/v2/durability/wal.cpp:619]: The `TEXT_*` cases were just inserted randomly between valid fallthrough cases. `LABEL_PROPERTY_INDEX_*` was previously grouped with the `EXISTENCE_CONSTRAINT_*` cases.

Luckily this did nothing to production code, atm only used in tests.

**Pull request merged**